.github/actions/windows_install/action.yml

```diff
@@ -37,31 +38,34 @@ runs:
         #
         # Install MS MPI
         ./msmpisetup.exe -unattend
-        sleep 5
+        sleep 20
         msiexec /quiet /i msmpisdk.msi
+        sleep 1
+        echo "MS MPI runtime and SDK installed"
+      shell: powershell
+    - name: Setup MPI
+      run: |
+        #refreshenv
+        $MSMPI_INC="C:\\Program Files (x86)\\Microsoft SDKs\\MPI\\Include\\"
+        $MSMPI_LIB64="C:\\Program Files (x86)\\Microsoft SDKs\\MPI\\Lib\\x64\\"
+        $PATH="${Env:Path};C:\\Program Files\\Microsoft MPI\\Bin"
+        [System.Environment]::SetEnvironmentVariable("Path",$PATH)
+        echo "Path=${PATH}" >> $GITHUB_ENV
```

Comment:
This is the line that is supposed to ensure that the system variable is set between shells. Maybe it should be:
```
echo "Env:Path=${PATH}" >> $GITHUB_ENV
```
?

Reply:
Yes, this is a bit tricky. I've now switched to another variant according to https://stackoverflow.com/questions/60169752/how-to-update-the-path-in-a-github-action-workflow-file-for-a-windows-latest-hos

Reply:
Beware that question was asked in 2011 and they seem to have changed method since then. They detected a security vulnerability in the envionment. I'm not sure when exactly but this link:
https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/
makes me think it must have been very recent.

The doc is here:
https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#environment-files

Reply:
Never mind, I just realised they say that in one of the answers ^^

Reply:
Hm... Now the installation and mpiexec is running, but I still get an error on the tests. Locally on my machine everything works, so could be a problem or a restriction with github actions related to network?

Reply:
Do you mind if I add a couple of prints to try and see what actually failed?

I'm definitely worried that it's just that the spawned process isn't in the right shell though :/

Reply:
Of course, please go ahead. I guess I leave this whole PR now to you again, since I'm at the end of my capabilities ^^.

Reply:
The error is `(<class 'KeyError'>, KeyError('MSMPI_INC'), <traceback object at 0x00000212F4802148>)`
so I think it's just a question of keeping `$MSMPI_INC` and `$MSMPI_LIB64` in the path

Reply:
So how was this problem solved?

Reply:
The problem was solved by using the command discussed above to keep `$MSMPI_INC` and `MSMPI_LIB64` in the environment variables